fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java

```diff
@@ -651,4 +668,172 @@ public static long analyzeLongProp(Map<String, String> properties, String propKe
+                        .stream().map(String::trim).collect(Collectors.toList());
+                if (baseColumns.size() != parentColumns.size()) {
+                    throw new AnalysisException(String.format("invalid foreign key constraint:%s," +
+                            " columns' size does not match", foreignKeyConstraintDesc));
+                }
+
+                String[] parts = tablePath.split("\\.");
+                String catalogName = InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME;
+                String dbName = db.getFullName();
+                String parentTableName = "";
+                if (parts.length == 3) {
+                    catalogName = parts[0];
+                    dbName = parts[1];
+                    parentTableName = parts[2];
+                } else if (parts.length == 2) {
+                    dbName = parts[0];
+                    parentTableName = parts[1];
+                } else if (parts.length == 1) {
+                    parentTableName = parts[0];
+                } else {
+                    throw new AnalysisException(String.format("invalid foreign key constraint:%s," +
+                            "table path is invalid", foreignKeyConstraintDesc));
+                }
+
+                if (!GlobalStateMgr.getCurrentState().getCatalogMgr().catalogExists(catalogName)) {
+                    throw new AnalysisException(String.format("catalog: %s do not exist", catalogName));
+                }
+                Database parentDb = GlobalStateMgr.getCurrentState().getMetadataMgr().getDb(catalogName, dbName);
+                if (parentDb == null) {
+                    throw new AnalysisException(String.format("catalog: %s, database: %s do not exist", catalogName, dbName));
+                }
+                Table parentTable =
+                        GlobalStateMgr.getCurrentState().getMetadataMgr().getTable(catalogName, dbName, parentTableName);
+                if (parentTable == null) {
+                    throw new AnalysisException(String.format("catalog:%s, database: %s, table:%s do not exist",
+                            catalogName, dbName, parentTableName));
+                }
+
+                if (!baseTable.isLocalTable()) {
+                    throw new AnalysisException("now do not support add foreign key on external table");
+                }
+
+                OlapTable baseOlapTable = (OlapTable) baseTable;
+                if (!baseColumns.stream().allMatch(baseOlapTable::containColumn)) {
+                    throw new AnalysisException(String.format("some columns of:%s do not exist in table:%s",
+                            baseColumns, baseOlapTable.getName()));
+                }
+
+                OlapTable parentOlapTable = (OlapTable) parentTable;
+                if (!parentColumns.stream().allMatch(parentOlapTable::containColumn)) {
+                    throw new AnalysisException(String.format("some columns of:%s do not exist in table:%s",
```

Comment:
`some columns of:%s do not exist in the parent table` ?

Reply:
done